src/Functions/extractTimeZoneFromFunctionArguments.cpp

```diff
@@ -65,7 +75,15 @@ const DateLUTImpl & extractTimeZoneFromFunctionArguments(const ColumnsWithTypeAn
     else
     {
         if (arguments.size() <= datetime_arg_num)
-            return DateLUT::instance();
+        {
+            if (!user_default_time_zone.empty())
+            {
+                DateLUT::setDefaultTimezone(user_default_time_zone);
```

Comment:
It is better to instantiate `DateLUT` of the specified timezone instead of setting it timezone as default - that would affect all queries of all users.